src/common.h

```diff
@@ -24,10 +24,30 @@
 #define __ALIGNED(x) __attribute__((aligned(x)))
 #elif defined(_MSC_VER)
 #define __ALIGNED(x) __declspec(align(x))
-#else /* unspported compilers */
+#else /* unsupported compilers */
 #define __ALIGNED(x)
 #endif
 
+/* Packed macro */
+#if defined(__GNUC__) || defined(__clang__)
```

Comment:
Rewrite as following:
```c
#if defined(__GNUC__) || defined(__clang__)
#define PACKED_STRUCT(name) struct __attribute__((packed)) name
#elif defined(_MSC_VER)
#define PACKED_STRUCT(name) \
    __pragma(pack(push, 1)) struct name __pragma(pack(pop))
#endif
```

Then this can be used like so:
```c
typedef PACKED_STRUCT(/*anonymous*/) { short a; int b } my_struct_t;
```